docs/proposals/release-notes.md

```diff
@@ -0,0 +1,174 @@
+
+<!-- TAG RELEASE_LINK, added by the munger automatically -->
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.2/docs/proposals/release-notes.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Kubernetes Release Notes
+[djmm@google.com](mailto:djmm@google.com)<BR>
+Last Updated: 2016-3-21
+
+# Objective
+
+Define a process and design tooling for collecting, arranging and publishing
+release notes for Kubernetes releases, automating as much of the process as
+possible. 
+
+The goal is to introduce minor changes to the development workflow
+in a way that is mostly frictionless and allows for the capture of release notes
+as PRs are submitted to the repository.
+
+This direct association of release notes to PRs captures the intention of
+release visibility of the PR at the point an idea is submitted upstream.
+The release notes can then be more easily collected and published when the
+release is ready.
+
+# Background
+
+## The Problem
+
+Release notes are often an afterthought and clarifying and finalizing them
+is often left until the very last minute at the time the release is made.
+This is usually long after the feature or bug fix was added and is no longer on
+the mind of the author.  Worse, the collecting and summarizing of the
+release is often left to those who may know little or nothing about these
+individual changes!
+
+Writing and editing release notes at the end of the cycle can be a rushed,
+interrupt-driven and often stressful process resulting in incomplete,
+inconsistent release notes often with errors and omissions.
+
+## The (general) Solution
+
+Like most things in the development/release pipeline, the earlier you do it,
+the easier it is for everyone and the better the outcome.  Gather your release
+notes earlier in the development cycle, at the time the features and fixes are
+added.  
+
+This concept was first implemented at Google for the Borg team to
+capture release notes at submission time.  Since then the concept has
+been adopted widely and used successfully across Google.
+
+### Then why not just list *every* change that was submitted, CHANGELOG-style?
+
+On larger projects like Kubernetes, showing every single change (PR) would mean
+hundreds of entries.  The goal is to highlight the major changes for a release.
+
+# Collection Design
+
+1. `hooks/prepare-commit-msg` and `hooks/commit-msg` provide commit templates
+  that capture the release-visiblity intention of that PR with a field inserted
+  into the commit message:
+
+  ```
+  # **Should this commit be a RELEASE NOTE? Choose one or DELETE the line.
+  # **Specify "breaking" for flag changes and the like
+  RELEASE NOTE: std | breaking
+  ```
+
+1. This field in the commit message acts as a trigger during the process of
```

Comment:
Some thoughts:
* Can every one of those sections be auto-generated from PR headings?  Or will some of those sections be more likely free-form text authored by the release owner(s)?
* Will every alpha, beta and minor release have those sections?  My thought is that only major (1.1, 1.2) will have all of those.
* release note section fatigue.  We want to introduce something to the workflow that is easy/intuitive to use.  We can include guidance in the form of commit message comments, but we are working with limited space there and should aim for brevity / intuitive design anyway.  The tyranny of choice.
* Maybe some but not all of the suggested labels/types here?